Framework/Core/src/DataAllocator.cxx

```diff
@@ -251,5 +251,15 @@ Output DataAllocator::getOutputByBind(OutputRef&& ref)
   O2_BUILTIN_UNREACHABLE();
 }
 
+bool DataAllocator::isValid(OutputSpec const& query)
```

Comment:
How about `isAllowed`?

Reply:
Also, shouldn't this use "Output" rather than `OutputSpec`?

Reply:
`isAllowed` is as good as `isValid`, can change this. For the query type I was also thinking about `Output` but then I wanted to have this close to what one adds to the output specs, and I wanted to use `ConcreteDataTypeMatcher`. Probably not a good idea, as one has to specify `Output` with all three coordinates anyhow in the API methods. So why not doing this consistently in the check. You are right.

Reply:
the code is adapted acordingly

Reply:
Depends what the purpose of the API is. If it is to check if a given message is a valid / allowed one, then ConcreteDataTypeMatcher / Output is the correct thing to have. If the goal is to check whether a given rule is among the ones present, then OutputSpec is what to use. Regarding the name, I guess my comment was more about "isValid" not referring to the DataAllocator itself, but to the passed object. `allowsFor` maybe it's a better name for what I meant.